libecl/src/ecl_grid.c

```diff
@@ -2202,19 +2211,20 @@ static ecl_grid_type * ecl_grid_alloc_GRDECL_data__(ecl_grid_type * global_grid
                                                     int lgr_nr) {
 
   ecl_grid_type * ecl_grid = ecl_grid_alloc_empty(global_grid , dualp_flag , nx,ny,nz,lgr_nr,true);
+  if (ecl_grid) {
```

Comment:
What about rather flipping this around, and just return early? I think this makes it easier to read.